### PR TITLE
MM-12389 Update segment library to v3

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Updates segment analytics-go library from v2 to v3 and patches code to use the new Api.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-12389

### CODE REVIEW

**Comment by streamer45** [comment]
Interesting that Jenkins is failing on this gorilla/handlers dependency which has nothing to do with the PR. I don't know much about it but it looks like it was removed in fdf9c3e218f33d8355209f54aa36853a027c2650.
I simply ran ```go mod tidy``` and ```go mod vendor``` as part of updating the segment library and it removed that too.

**Comment by jespino** [comment]
@streamer45 The problem here is you are executing `go mod tidy` without having the `enterprise` repo, so certain things are removed because you don't have the code for it. I think the idea here should be to remove the lines that are related to handlers.

**Comment by streamer45** [comment]
@jespino thanks for the info, now it makes more sense. I could keep ```gorilla/handlers``` as a dependency so that it doesn't break enterprise build which I am guessing relies on that, but I would have to modify ```go.sum``` and ```go.mod``` by hand to only update the ```segment/analytics-io``` module. Is that what you are proposing?

**Comment by jespino** [comment]
@streamer45 Yes, that is my propose, I can check by myself the result of the `go tidy` locally ones you have made the changes :)

File: vendor/modules.txt [hunk 1]
```diff
@@ -48,8 +48,6 @@ github.com/golang/protobuf/ptypes/timestamp
 github.com/google/btree
 # github.com/google/uuid v1.1.1
 github.com/google/uuid
-# github.com/gorilla/handlers v1.4.0
```
Comment (jespino) [inline vendor/modules.txt]: I guess this shouldn't be removed.

Comment (streamer45) [reply vendor/modules.txt]: Yep, I overlooked. Fixed now.

**Pull request merged**